.github/workflows/reusable-build-docker-image.yaml

```diff
@@ -0,0 +1,221 @@
+      # what variables we have available to this workflow.
+      IMAGE_NAME: ${{ needs.get-job-flags.outputs.IMAGE_NAME }}
+      REPO_NAME: ${{ needs.get-job-flags.outputs.REPO_NAME }}
+      DOCLINTTEST_TAG: ${{ needs.get-job-flags.outputs.DOCLINTTEST_TAG }}
+      DOCLINTTEST_IMAGE: ${{ needs.get-job-flags.outputs.DOCLINTTEST_IMAGE }}
+
+  # Build the base GeoIPS doclinttest image
+  # This image is used in all subsequent tests
+  # It is only built for the main GeoIPS package but is used by all repositories
+  #
+  # Within this image, the GeoIPS package is located in /packages/geoips and is
+  # installed using `pip install .[doc,lint,test]`. Scripts from geoips can be called
+  # from /packages/geoips.
+  build_doclinttest_image:
+    runs-on: ${{ vars.RUNNER_DOCKERBUILD || vars.RUNNER }}
+    needs: get-job-flags
+    if: >
+      ! cancelled() &&
+      vars.USE_CONDA_FOR_ACTIONS != 'true' &&
+      github.event.repository.name == 'geoips'
+    env:
+      IMAGE_NAME: ${{ needs.get-job-flags.outputs.IMAGE_NAME }}
+      BUILDCACHE_IMAGE: ${{ needs.get-job-flags.outputs.IMAGE_NAME }}:buildcache
+      DOCLINTTEST_IMAGE: ${{ needs.get-job-flags.outputs.DOCLINTTEST_IMAGE }}
+    # Can't write to registry without this
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup BuildX
+        if: ${{ vars.USE_DOCKER_BUILDX_ACTION == 'true' }}
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to the Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ vars.DOCKER_REGISTRY }}
+          username: ${{ vars.DOCKER_REGISTRY_USER }}
+          password: ${{ secrets.docker_token }}
+
+      # This cache allows the second push below when on the default branch
+      - name: Cache Docker layers
+        uses: actions/cache@v4
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+
+      - name: Create tags
+        run: |
+          tags="${DOCLINTTEST_IMAGE},${BUILDCACHE_IMAGE}"
+          echo "GITHUB REF: ${GITHUB_REF}"
+          echo "Default Branch: refs/heads/${{ github.event.repository.default_branch }}"
+          echo "runner os: ${{ runner.os }}"
+          echo "github sha: ${{ github.sha }}"
+          echo "build cache image: ${{ env.BUILDCACHE_IMAGE }}"
+          # Create stable image on default branch
+          if [[ "${GITHUB_REF}" == "refs/heads/${{ github.event.repository.default_branch }}" ]]; then
+            stableorlatest_image="${IMAGE_NAME}:doclinttest-stable"
+            tags="${tags},${stableorlatest_image}"
+          # Also create stable image on tag
+          elif [[ "${GITHUB_REF}" == "refs/tags/"* ]]; then
+            stableorlatest_image="${IMAGE_NAME}:doclinttest-stable"
+            tags="${tags},${stableorlatest_image}"
```

Comment:
This is the only part I changed - otherwise I just pulled this directly from resuable-doc-test.yaml.

Previously was building stable image on main, now build stable image on tag.  Note this will build on EVERY tag, and replace the stable image with the most recently tagged tag.  Not sure that is 100% what we want, but general release process that should work.

Reply:
This sounds fine for now. We probably want this to only produce `stable` on tags with a specific format, though, right?